eng/versioning/pom_file_version_scanner.ps1

```diff
@@ -510,13 +519,35 @@ $ArtifactsPerSD = [ordered]@{};
 Get-ArtifactsList-Per-Service-Directory $ArtifactsPerSD
 
 # Create one dependency hashtable for libraries we build (the groupIds will make the entries unique) and
-# one hash for external dependencies
+# one hash for external dependencies. If there are errors building the dependency hash, output those and
+# exit. There's no point in scanning the pom files if any of the files have invalid or duplicate entries.
 $libHash = @{}
-Build-Dependency-Hash-From-File $libHash $Path\eng\versioning\version_client.txt $false
-Build-Dependency-Hash-From-File $libHash $Path\eng\versioning\version_data.txt $false
+$verClientFile = Join-Path $Path "eng/versioning/version_client.txt"
+Build-Dependency-Hash-From-File $libHash $verClientFile $false
+if ($script:FoundError)
+{
+    Write-Error-With-Color "There were errors encountered building the dependency hash from version_client.txt. Please fix errors and run the script again."
+    Write-Error-With-Color "This script can be run locally from the root of the repo. .\eng\versioning\pom_file_version_scanner.ps1"
```

Comment:
maybe jam this in a function

Reply:
At this point, I'm just going to leave this as is.